sdks/python/apache_beam/runners/dataflow/native_io/iobase_test.py

```diff
@@ -0,0 +1,227 @@
+
+
+class TestReaderPosition(unittest.TestCase):
+
+  def test_invalid_concat_position_type(self):
+    with self.assertRaises(AssertionError):
+      ReaderPosition(concat_position=1)
+
+  def test_valid_concat_position_type(self):
+    ReaderPosition(concat_position=ConcatPosition(None, None))
+
+
+class TestConcatPosition(unittest.TestCase):
+
+  def test_invalid_position_type(self):
+    with self.assertRaises(AssertionError):
+      ConcatPosition(None, position=1)
+
+  def test_valid_concat_position_type(self):
+    ConcatPosition(None, position=ReaderPosition())
+
+
+class TestDynamicSplitRequest(unittest.TestCase):
+
+  def test_invalid_progress_type(self):
+    with self.assertRaises(AssertionError):
+      DynamicSplitRequest(progress=1)
+
+  def test_valid_progress_type(self):
+    DynamicSplitRequest(progress=ReaderProgress())
+
+
+class TestDynamicSplitResultWithPosition(unittest.TestCase):
+
+  def test_invalid_stop_position_type(self):
+    with self.assertRaises(AssertionError):
+      DynamicSplitResultWithPosition(stop_position=1)
+
+  def test_valid_stop_position_type(self):
+    DynamicSplitResultWithPosition(stop_position=ReaderPosition())
+
+
+class TestNativeSink(unittest.TestCase):
+
+  def test_writer_method(self):
+    native_sink = NativeSink()
+    self.assertRaises(NotImplementedError, native_sink.writer)
+
+  def test_repr_method(self):
+    class FakeSink(NativeSink):
+      """A fake sink modeled after BigQuerySink, which inherits from
+      NativeSink."""
+
+      def __init__(self, validate=False, dataset=None, project=None,
+                   schema=None, create_disposition='create',
+                   write_disposition=None, coder=None):
+        self.validate = validate
+
+    fake_sink = FakeSink()
+    self.assertEqual(fake_sink.__repr__(), "<FakeSink ['validate=False']>")
+
+
+class TestNativeSinkWriter(unittest.TestCase):
+
+  def setUp(self):
+    self.native_sink_writer = NativeSinkWriter()
+
+  def test_enter_method(self):
+    self.assertRaises(NotImplementedError, self.native_sink_writer.__enter__)
+
+  def test_exit_method(self):
+    with self.assertRaises(NotImplementedError):
+      self.native_sink_writer.__exit__(None, None, None)
+
+  def test_takes_windowed_values_property(self):
+    self.assertEqual(self.native_sink_writer.takes_windowed_values, False)
+
+  def test_Write_method(self):
+    self.assertRaises(NotImplementedError, self.native_sink_writer.Write, None)
+
+
+class Test_NativeWrite(unittest.TestCase):
+
+  def setUp(self):
+    self.native_sink = NativeSink()
+    self.native_write = _NativeWrite(self.native_sink)
+
+  def test_expand_method_pcollection_errors(self):
+    with self.assertRaises(error.TransformError):
+      self.native_write.expand(None)
+    with self.assertRaises(error.TransformError):
+      pcoll = pvalue.PCollection(pipeline=None)
+      self.native_write.expand(pcoll)
+
+  def test_expand_method(self):
```

Comment:
I think PDone is going to be removed in the near future -- https://github.com/apache/beam/pull/2394 -- so let's not add this test.